### ISSUE TITLE
Bug: `no-unsafe-finally` misses labeled `continue` targeting a loop outside `finally`

### ISSUE THREAD

**Issue by electrohyun** [eslint/eslint#21314]
### Environment

Node version: v22.16.0
npm version: v11.17.0
Local ESLint version: v10.9.1
Global ESLint version: None
Operating System: macOS 26.5 (Apple Silicon, arm64)

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
    {
        rules: {
            "no-unsafe-finally": "error",
        },
    },
];
```
</details>

```js
function foo() {
  outer: for (let i = 0; i < 1; i++) {
    try {
      return 1;
    } finally {
      for (let j = 0; j < 1; j++) {
        continue outer; // Not reported
      }
    }
  }

  return 2;
}

foo(); // 2
```

### What did you expect to happen?

The labeled `continue` statement should be reported as unsafe.

### What actually happened?

No error was reported.

### Link to Minimal Reproducible Example

[Playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVuc2FmZS1maW5hbGx5OiBcImVycm9yXCIsIG5vLXVudXNlZC12YXJzOiBcIm9mZlwiICovXG5cbmZ1bmN0aW9uIGZvbygpIHtcbiAgb3V0ZXI6IGZvciAobGV0IGkgPSAwOyBpIDwgMTsgaSsrKSB7XG4gICAgdHJ5IHtcbiAgICAgIHJldHVybiAxO1xuICAgIH0gZmluYWxseSB7XG4gICAgICBmb3IgKGxldCBqID0gMDsgaiA8IDE7IGorKykge1xuICAgICAgICBjb250aW51ZSBvdXRlcjsgLy8gTm90IHJlcG9ydGVkXG4gICAgICB9XG4gICAgfVxuICB9XG5cbiAgcmV0dXJuIDI7XG59XG5cbmZvbygpOyAvLyAyIiwib3B0aW9ucyI6eyJydWxlcyI6eyJjb25zdHJ1Y3Rvci1zdXBlciI6WyJlcnJvciJdLCJmb3ItZGlyZWN0aW9uIjpbImVycm9yIl0sImdldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tYXN5bmMtcHJvbWlzZS1leGVjdXRvciI6WyJlcnJvciJdLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6WyJlcnJvciJdLCJuby1jbGFzcy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29tcGFyZS1uZWctemVybyI6WyJlcnJvciJdLCJuby1jb25kLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3RhbnQtYmluYXJ5LWV4cHJlc3Npb24iOlsiZXJyb3IiXSwibm8tY29uc3RhbnQtY29uZGl0aW9uIjpbImVycm9yIl0sIm5vLWNvbnRyb2wtcmVnZXgiOlsiZXJyb3IiXSwibm8tZGVidWdnZXIiOlsiZXJyb3IiXSwibm8tZGVsZXRlLXZhciI6WyJlcnJvciJdLCJuby1kdXBlLWFyZ3MiOlsiZXJyb3IiXSwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjpbImVycm9yIl0sIm5vLWR1cGUtZWxzZS1pZiI6WyJlcnJvciJdLCJuby1kdXBlLWtleXMiOlsiZXJyb3IiXSwibm8tZHVwbGljYXRlLWNhc2UiOlsiZXJyb3IiXSwibm8tZW1wdHkiOlsiZXJyb3IiXSwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLWVtcHR5LXBhdHRlcm4iOlsiZXJyb3IiXSwibm8tZW1wdHktc3RhdGljLWJsb2NrIjpbImVycm9yIl0sIm5vLWV4LWFzc2lnbiI6WyJlcnJvciJdLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOlsiZXJyb3IiXSwibm8tZmFsbHRocm91Z2giOlsiZXJyb3IiXSwibm8tZnVuYy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZ2xvYmFsLWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbXBvcnQtYXNzaWduIjpbImVycm9yIl0sIm5vLWludmFsaWQtcmVnZXhwIjpbImVycm9yIl0sIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjpbImVycm9yIl0sIm5vLWxvc3Mtb2YtcHJlY2lzaW9uIjpbImVycm9yIl0sIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLW5ldy1uYXRpdmUtbm9uY29uc3RydWN0b3IiOlsiZXJyb3IiXSwibm8tbm9ub2N0YWwtZGVjaW1hbC1lc2NhcGUiOlsiZXJyb3IiXSwibm8tb2JqLWNhbGxzIjpbImVycm9yIl0sIm5vLW9jdGFsIjpbImVycm9yIl0sIm5vLXByb3RvdHlwZS1idWlsdGlucyI6WyJlcnJvciJdLCJuby1yZWRlY2xhcmUiOlsiZXJyb3IiXSwibm8tcmVnZXgtc3BhY2VzIjpbImVycm9yIl0sIm5vLXNlbGYtYXNzaWduIjpbImVycm9yIl0sIm5vLXNldHRlci1yZXR1cm4iOlsiZXJyb3IiXSwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOlsiZXJyb3IiXSwibm8tc3BhcnNlLWFycmF5cyI6WyJlcnJvciJdLCJuby10aGlzLWJlZm9yZS1zdXBlciI6WyJlcnJvciJdLCJuby11bmFzc2lnbmVkLXZhcnMiOlsiZXJyb3IiXSwibm8tdW5kZWYiOlsiZXJyb3IiXSwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXSwibm8tdW5yZWFjaGFibGUiOlsiZXJyb3IiXSwibm8tdW5zYWZlLWZpbmFsbHkiOlsiZXJyb3IiXSwibm8tdW5zYWZlLW5lZ2F0aW9uIjpbImVycm9yIl0sIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6WyJlcnJvciJdLCJuby11bnVzZWQtbGFiZWxzIjpbImVycm9yIl0sIm5vLXVudXNlZC1wcml2YXRlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1hc3NpZ25tZW50IjpbImVycm9yIl0sIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6WyJlcnJvciJdLCJuby11c2VsZXNzLWNhdGNoIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl0sIm5vLXdpdGgiOlsiZXJyb3IiXSwicHJlc2VydmUtY2F1Z2h0LWVycm9yIjpbImVycm9yIl0sInJlcXVpcmUteWllbGQiOlsiZXJyb3IiXSwidXNlLWlzbmFuIjpbImVycm9yIl0sInZhbGlkLXR5cGVvZiI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=)

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

<img width="1248" height="548" alt="Image" src="https://github.com/user-attachments/assets/a0fa64a0-ad65-4b2c-aea5-c31d31050de8" />

The `no-unsafe-finally` rule disallows `continue` statements that can override a `return` or `throw` in `try`/`catch`. However, using `continue` inside `finally` as in the example cancels the pending `return`, so it should be reported.

Although this could be the code author’s intention, I think it would be good for `no-unsafe-finally` to report this as unsafe.

That's because [the documentation](https://github.com/eslint/eslint/blob/main/docs/src/rules/no-unsafe-finally.md?plain=1#L8) also explains that

> control flow statements inside `try` and `catch` are overwritten, which is considered as unexpected behavior. 

And, I found that an invalid test case for the same structure was requested in https://github.com/eslint/eslint/pull/6000#issuecomment-215725091 10 years ago.

I thought it would be better to ask in an issue first rather than submit a PR. I’d appreciate your thoughts on this. Thank you!

<hr />

Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

**Comment by mdjermanovic** [comment]
Thanks for the report. I agree this is a bug, PR is welcome.
